biogeochem/EDPhysiologyMod.F90

```diff
@@ -408,135 +408,185 @@ subroutine phenology( currentSite, bc_in )
     !2) The leaves should not be on already
-    !3) There should have been at least on chilling day in the counting period.  
-    if (currentSite%ED_GDD_site > gdd_threshold)then
-       if (currentSite%is_cold) then
-          if (currentSite%ncd >= 1) then
-             ! NOTE(bja, 2015-01) should leafondate = model_day to be consistent with leaf off?
-             currentSite%is_cold = .false.      
-             currentSite%leafondate = t         ! record leaf on date   
-             if ( debug ) write(fates_log(),*) 'leaves on'
-          endif !ncd
-       endif !status
+    !3) There should have been at least one chilling day in the counting period.  
+    !   this prevents tropical or warm climate plants that are "cold-deciduous"
+    !   from ever re-flushing after they have reached their maximum age (thus
+    !   preventing them from competing
+
+    if ( (currentSite%cstatus == 1 .or. currentSite%cstatus == 0) .and. &
+         (currentSite%grow_deg_days > gdd_threshold) .and. &
+         (currentSite%nchilldays >= 1)                    .and. &
+         (dayssincecleafoff > ED_val_phen_mindayson))  then
+       currentSite%cstatus = 2     !alter status of site to 'leaves on'
+       currentSite%cleafondate = model_day_int  
+       if ( debug ) write(fates_log(),*) 'leaves on'
     endif !GDD
 
-    timesinceleafon = hlm_model_day - currentSite%leafondate
+
 
 
     !LEAF OFF: COLD THRESHOLD
     !Needs to:
     !1) have exceeded the number of cold days threshold
     !2) have exceeded the minimum leafon time.
     !3) The leaves should not be off already
-    !4) The day of the year should be larger than the counting period. 
-    !   (not sure if we need this/if it will break the restarting)
+    !4) The day of simulation should be larger than the counting period. 
+
     
-    if (ncolddays > ED_val_phen_ncolddayslim)then
-     if (timesinceleafon > ED_val_phen_mindayson)then
-       if (.not.currentSite%is_cold)then
-          currentSite%is_cold = .true.              ! now cold enough for leaf-off
-          currentSite%leafoffdate = hlm_model_day   ! record leaf off date   
-          if ( debug ) write(fates_log(),*) 'leaves off'
-       endif
-    endif
+    if ( (currentSite%cstatus == 2)             .and. &
+         (model_day_int > num_vegtemp_mem)      .and. &
+         (ncolddays > ED_val_phen_ncolddayslim) .and. &
+         (dayssincecleafon > ED_val_phen_mindayson) )then
+       
+       currentSite%grow_deg_days  = 0._r8          ! The equations for Botta et al
+                                                 ! are for calculations of 
+                                                 ! first flush, but if we dont
+                                                 ! clear this value, it will cause
+                                                 ! leaves to flush later in the year
+       currentSite%cstatus       = 1             ! alter status of site to 'leaves on'
```

Comment:
This is a comment typo correct?  Should it read `! alter status of site to 'leaves off'`?

Reply:
yes! it should alter status to 'leaves off'

good catch, that is confusing